client/src/mifare/desfire_crypto.c

```diff
@@ -251,7 +251,7 @@ static void xor(const uint8_t *ivect, uint8_t *data, const size_t len) {
     }
 }
 
-void cmac_generate_subkeys(desfirekey_t key) {
+void cmac_generate_subkeys(desfirekey_t key, MifareCryptoDirection direction) {
```

Comment:
The direction parameter was added here. since the xor'ing of the result during the mifare CBC function was screwing with the key generation needed for AN10922